src/spaces/continuous.jl

```diff
@@ -410,3 +356,222 @@ function Base.iterate(iter::PairIterator, i = 1)
     id1, id2 = p
     return (iter.agents[id1], iter.agents[id2]), i + 1
 end
+
+#######################################################################################
+# FMP Algorithm
+#######################################################################################
+export fmp_update_vel!, fmp_update_interacting_pairs!, FMP_Parameters, fmp_parameter_init
+
+"""
+    FMP_Parameters
+The parameters for the FMP model as defined in the [FMP paper](https://arxiv.org/abs/1909.05415). The helper function `fmp_parameter_init` is used to initialize the struct with default parameters. The default values in `fmp_parameter_init` generally reflect the parameterization from the original paper. The available FMP parameters are:
+- `rho`: Positive repulsive gradient; dictates the magnitude of the repulsive force as a function of inter-agent distance.
+- `rho_obstacle`: Positive repulsive gradient for objects; similar to `rho` but used for agent-object repulsive forces.
```

Comment:
This is a weird convention: you use "obstacle" in the name of the variable, but "object" when describing it. I think only one of the two terms should be used to avoid confusion.